lib/azure/armrest/configuration.rb

```diff
@@ -193,7 +189,10 @@ def validate_token_time(time)
       # a non-preview version that is not set in the future. Otherwise, just
       # just the most recent one.
       #
-      def set_provider_info(providers)
+      def set_provider_api_versions
+        # A lookup table for getting api-version strings per provider and service.
+        @provider_api_versions = Hash.new { |hash, key| hash[key] = {} }
```

Comment:
I'm just wondering if that will cause a warning in verbose mode. We can always revisit.